### ISSUE TITLE
Print glitch on exit

### ISSUE THREAD

**Issue by BenWiederhake** [bpython/bpython#722]
Let's assume the following file `calls_exit_2.py` in python3.  (Not sure if something similar happens for python2.)

```python3
import sys

def do_it():
    exit(2)

def and_print_something():
    print('Good bye, World!', file=sys.stderr)
    exit(2)
```

With this, I will show that bpython3 behaves a bit weird, what I assume to be two separate bugs.

### One: bpython3 prints the exit code as a tuple.

Input: `import calls_exit_2; calls_exit_2.do_it()`

Expected behavior: Either don't print anything and exit with exit code 2, or print some message like `Program exited with code 2.` and return to the bpython3 shell.  Both are fine.

Actual behavior: Exit with code 2, leaving the following visible on the screen:

```
bpython version 0.17 on top of Python 3.6.4rc1 /usr/bin/python3
>>> import calls_exit_2; calls_exit_2.do_it()
(2,)
```

Where does the `(2,)` come from?

### Two: bpython3 repeats the last input, if there is programmatic output on stderr.

Input: `import calls_exit_2; calls_exit_2.and_print_something()`

Expected behavior: Either only print `Goodbye, World!` and exit with exit code 2, or print some message like `Goodbye, World!\nProgram exited with code 2.` and return to the bpython3 shell.  Both are fine.

Actual behavior: Exit with code 2, leaving the following visible on the screen:

```
bpython version 0.17 on top of Python 3.6.4rc1 /usr/bin/python3
>>> import calls_exit_2; calls_exit_2.and_print_something()
>>> import calls_exit_2; calls_exit_2.and_print_something()
(2,)
```

Where does the second "input" line come from?

**Comment by BenWiederhake** [comment]
For reference, here are the outputs of just python3:

```
Python 3.6.4rc1 (default, Dec  6 2017, 10:08:29) 
[GCC 7.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import calls_exit_2; calls_exit_2.do_it()
```
(Exit code is 2, no `(2,)` output.)

```
Python 3.6.4rc1 (default, Dec  6 2017, 10:08:29) 
[GCC 7.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import calls_exit_2; calls_exit_2.and_print_something()
Good bye, World!
```
(Exit code is 2, no `(2,)` output or repeated "input".)

**Comment by BenWiederhake** [comment]
Oh, and the detour via a file is not necessary.

```
$ bpython3
bpython version 0.17 on top of Python 3.6.4rc1 /usr/bin/python3
>>> exit(2)
(2,)
$ echo $?
1
$ bpython3
bpython version 0.17 on top of Python 3.6.4rc1 /usr/bin/python3
>>> import sys; print('Goodbye, World!', file=sys.stderr); exit(2)
>>> import sys; print('Goodbye, World!', file=sys.stderr); exit(2)
(2,)
$ echo $?
1
```

Note that this reveals two more bugs:

### Three: Wrong exit code

Expected behavior: exit code 2

Actual behavior: exit code 1

### Four: Vanished output

Expected behavior: There is an output saying `Goodbye, World!`

Actual behavior: No such output.

This is probably due to some data race, and maybe difficult to fix.  I can imagine you're tempted to say "Eh, that's an edge case noone cares about!", but let me give you a counter-example: User does a library call, which decides it needs to abort the entire process.  Trying to be helpful, it outputs some info about what happened, so that the user isn't baffled by the sudden abort.  Swallowing that info definitely causes some headaches then.

Some of these "bugs" might be different symptoms of the same root cause.  That's why I write all four "bugs" in the same issue.

**Comment by thomasballinger** [comment]
Thanks for submitting these @BenWiederhake. My thoughts on each:

1. whoops, this has something to do with not moving the args of the SystemExit exception around correctly. It's fixed if I add a star top unpack the args in https://github.com/bpython/bpython/blob/ab6f17846e858e3f4e77a4fe00cba44275f37786/bpython/curtsiesfrontend/coderunner.py#L186 but I need to test this with Python 2/3, didn't something about exception args change between these?

2 + 4. Hm, I remember writing this special case code: the issue was not knowing what line the cursor was on. I think this has to do with https://github.com/bpython/bpython/blob/ab6f17846e858e3f4e77a4fe00cba44275f37786/bpython/curtsiesfrontend/repl.py#L1317
I agree swallowing stderr on exit is not good at all, your 

3. hopefully fixed by 1

**Comment by sebastinas** [comment]
Fixed in 68df142edb1f9adb0b348374277211e1802de39c.
